### ISSUE TITLE
Allow dynamic client token path assembly

### ISSUE THREAD

**Issue by jeffcarp** [jeffcarp/braintree-angular#30]
Via @gigster-starchup in https://github.com/jeffcarp/braintree-angular/issues/23#issuecomment-149236361

Most likely involves turning `clientTokenPath` from a constant into a service.

**Comment by simonasdev** [comment]
Also we could just pass absolute/relative url right into the function `getClientToken(url, params)` or `getClientToken(params, url)` and fallback from missing parameter to service/constant value.

**Comment by dfibuch** [comment]
So right now there is no way to set the clientTokenPath anywhere else apart from when declaring the app? The first part of my URL is stored in a configuration service which is then used to build up all other URLs, which means I need to bring this service in to build the token path.

**Comment by jeffcarp** [comment]
@dfibuch correct, that is not currently supported. I hope to get to this at some point but before then, PR's are welcome!

**Comment by dfibuch** [comment]
Thanks for your response. I have made my own slight modifications which allow me to pass in the tokenUrl as part of the directive so that I can set it in the controller via the ConfigurationService I have. This then eliminates the need to set the clientTokenPath at all. Not sure if this is something you'd be looking for.

**Comment by jeffcarp** [comment]
@dfibuch that sounds like a great solution, if you have the time could you PR it?
